models/DataObject/ClassDefinition/Data/Relations/ManyToManyRelationTrait.php

```diff
@@ -58,4 +60,20 @@ public function save(Localizedfield|AbstractData|\Pimcore\Model\DataObject\Objec
 
         parent::save($object, $params);
     }
+
+    protected function filterUnpublishedElements(mixed $data): array
+    {
+        if (DataObject::doHideUnpublished() && is_array($data)) {
```

Comment:
codewise LGTM (haven't tested it yet)

just a micro-optimization to reduce one `is_array`

```suggestion
        if (!is_array($data)){ return [];}
        if (DataObject::doHideUnpublished()) {
```
and at the bottom replacing with 
`return $data`

Reply:
@kingjia90 good point, just pushed it (was just copy & pasting it 😉 ) Thanks!